cvmfs/signature.cc

```diff
@@ -290,6 +290,47 @@ bool SignatureManager::LoadPublicRsaKeys(const string &path_list) {
 }
 
 
+std::string SignatureManager::GenerateKeyText(RSA *pubkey) {
+  if (!pubkey) {return "";}
+
+  BIO *bp = BIO_new(BIO_s_mem());
+  if (bp == NULL) {
+    LogCvmfs(kLogSignature, kLogDebug | kLogSyslogErr, "Failed to allocate"
+             " memory for pubkey");
+    return "";
+  }
+  if (!PEM_write_bio_RSA_PUBKEY(bp, pubkey)) {
+    LogCvmfs(kLogSignature, kLogDebug | kLogSyslogErr, "Failed to write"
+             " pubkey to memory");
+    return "";
+  }
+  char *bio_pubkey_text;
+  long bytes = BIO_get_mem_data(bp, &bio_pubkey_text);
+  std::string bio_pubkey_str(bio_pubkey_text, bytes);
+  BIO_free(bp);
+
+  return bio_pubkey_str;
+}
+
+
+std::string SignatureManager::GetActivePubkeys() {
+  std::string pubkeys;
+  for (std::vector<RSA *>::const_iterator it = public_keys_.begin();
+       it != public_keys_.end();
+       it++) {
+    pubkeys += GenerateKeyText(*it);
+  }
+  if (certificate_) {
+    // Increases the refcount of the rsa object.
+    RSA *rsa = EVP_PKEY_get1_RSA(X509_get_pubkey(certificate_));
```

Comment:
The public key of the certificate shouldn't matter, actually.  It's not used for the whitelist verification.